server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/AbstractSignificanceHeuristicTests.java

```diff
@@ -54,16 +51,18 @@
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
 import static org.hamcrest.Matchers.greaterThanOrEqualTo;
-import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.lessThan;
-import static org.hamcrest.Matchers.lessThanOrEqualTo;
 
-public class SignificanceHeuristicTests extends ESTestCase {
+public abstract class AbstractSignificanceHeuristicTests extends ESTestCase {
```

Comment:
I think a little javadoc here to explain what you'll get by implementing this class, and what you still need to hand roll for each heuristic would be helpful.  In general, I'm trying to avoid undocumented abstract classes.